OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -2719,6 +2719,81 @@ public static DreamValue NativeProc_sorttext(DreamObject instance, DreamObject u
             return new DreamValue(0);
         }
 
+        /// <summary> Global turn() proc </summary>
+        /// <remarks> Take note that this turn proc is a counterclockwise rotation unlike the rest </remarks>
+        [DreamProc("turn")]
+        [DreamProcParameter("Dir", Type = DreamValueType.Float)]
+        [DreamProcParameter("Angle", Type = DreamValueType.Float)]
+        public static DreamValue NativeProc_turn(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
+            DreamValue dirArg = arguments.GetArgument(0, "dir");
+            DreamValue angleArg = arguments.GetArgument(1, "angle");
+
+            // Handle an invalid angle, defaults to 0
+            if (!angleArg.TryGetValueAsFloat(out float angle)) {
+                angle = 0;
+            }
+
+            // If Dir is actually an icon, call /icon.Turn
+            if (dirArg.TryGetValueAsDreamObjectOfType(ObjectTree.Icon, out var icon)) {
+                // Clone icon here since it's specified to return a new one
+                DreamObject clonedIcon = DreamMetaObjectIcon.CloneIcon(ObjectTree, icon);
+                return DreamProcNativeIcon._NativeProc_TurnInternal(clonedIcon, usr, angle);
+            }
+
+            // If Dir is actually a matrix, call /matrix.Turn
+            if (dirArg.TryGetValueAsDreamObjectOfType(ObjectTree.Matrix, out var matrix)) {
+                // Clone matrix here since it's specified to return a new one
+                DreamObject clonedMatrix = DreamMetaObjectMatrix.MatrixClone(ObjectTree, matrix);
+                return DreamProcNativeMatrix._NativeProc_TurnInternal(clonedMatrix, usr, angle);
+            }
+
+            if (!dirArg.TryGetValueAsInteger(out int possibleDir)) {
+                throw new ArgumentException($"Invalid Dir for Turn: \"{dirArg.ToString()}\"");
```

Comment:
A non-number `Dir` defaults to 0 in BYOND.

Reply:
Fixed.